packages/lsd-react/src/components/Dropdown/Dropdown.tsx

```diff
@@ -43,33 +45,50 @@ export const Dropdown: React.FC<DropdownProps> & {
   disabled = false,
   supportingText,
   triggerLabel,
-
   value = [],
   onChange,
   options = [],
   multi = false,
   variant = 'outlined',
+  isOpen,
+  onToggle,
   ...props
 }) => {
   const commonProps = useCommonProps(props)
   const containerRef = useRef<HTMLDivElement>(null)
-  const [open, setOpen] = useState(false)
+  const isControlled = isOpen !== undefined
+  const [openState, setOpenState] = useState(false)
 
   const { select, isSelected, selected } = useSelect(options, value, {
     multi,
     onChange,
     onDone: () => {
-      setOpen(false)
+      setOpenState(false)
     },
   })
 
+  const handleToggle = (open: boolean) => {
+    if (isControlled) {
+      onToggle && onToggle(open)
+    } else {
+      setOpenState(open)
+    }
+  }
+
   const onTrigger = () => {
-    !disabled && setOpen((value) => !value)
+    if (disabled) return
+
+    handleToggle(!openState)
   }
 
   useEffect(() => {
-    if (disabled && open) setOpen(false)
-  }, [open, disabled])
+    if (disabled && openState && !isControlled) setOpenState(false)
+  }, [open, disabled, isControlled])
```

Comment:
I think there should be linting to prevent situations like this. I should also note the `open` name came from before this PR - I just forgot to remove it here.